### ISSUE TITLE
one question

### ISSUE THREAD

**Issue by syf980302** [rohin-dasari/Tensorflow-Knowledge-Distillation#2]
Hello, I have a question for you. I want to ask how you call out the architecture of the teacher model. I tried many methods here, but I couldn't get it. Thank you very much.

**Comment by rohin-dasari** [comment]
What do you mean by "call out the architecture of the teacher model". Do you mean how its defined?

**Comment by syf980302** [comment]
Yes. I see that your readme file has a structural distribution diagram of the teacher model, but I don't know how to get it, and how to train the teacher model.

**Comment by rohin-dasari** [comment]
The model itself is defined in the `teacherModel` function of `distiller.py` (line 235). The teacher is trained in the `trainTeacher` method in the `Distiller` class (line 90). Both the teacher and student models are passed as parameters to the `Distiller` class. If you want to redefine the model or swap it out with a pre-trained model from tensorflow, you should just be able to create your own model and pass it as an argument to the `Distiller` object.

**Comment by syf980302** [comment]
I see what you mean. My question is how the teacher model is trained.

**Comment by rohin-dasari** [comment]
The teacher model is trained in the `train_teacher` method of the `Distiller` class. Are you asking how it works in the code or are you asking about how the teacher model works conceptually in knowledge distillation?

If you just want to train the teacher model you should be able to do something like this:
```python
distiller = Distiller(
    teacher=teacher
    student=tf.keras.models.clone_model(teacher),
    x_train=x_train,
    y_train=y_train,
    x_test=x_test,
    y_test=y_test,
    temp=4, 
    epochs=EPOCHS,
    task_balance=0.8)

dis.trainTeacher()
```

The training stats for the teacher are stored in `distiller.history['teacher']` and you can obtain the model after training with `distiller.teacher`

**Comment by syf980302** [comment]
What I want to ask is how it works in the code？"The teacher model is trained in the method of the class." 
I don't know what that means。

**Comment by rohin-dasari** [comment]
I'm not sure I understand what "how it works in the code" means. You could be asking for low-level implementation details, or high-level conceptual overview, or specefic design choices, etc. I have no way of knowing what I should really explain. 

The teacher model is essentially trained the same way you would train any model. It uses does a forward pass on each batch of data, computes the gradients, and back-propagates through the network to update the teacher model. The model uses categorical crossentropy as its loss function and tracks categorical accuracy as its primary performance metric. The actual logic for training the network is very similar to the logic found here: https://www.tensorflow.org/guide/keras/writing_a_training_loop_from_scratch#using_the_gradienttape_a_first_end-to-end_example

The main point is that training the teacher is not dependent at all on the student or knowledge distillation at all. It's just training a regular model. The logic for this is in the `teacherModel` method in the `Distiller` class. You can read through that method (which starts on line 90 of `distiller.py`) to get a sense of where each of the things I mentioned is happening in the code. After the teacher model is trained, we generate the predictions produced by the teacher model and use the targets to train the student model.
